modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+		_inputValue = "";
+		_label = "";
+		_locator = null;
+		_selectedItems = new ArrayList<>();
+		_sourceItems = new ArrayList<>();
+		_valid = true;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
+		props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
+		props.put("disabled", _disabled);
+		props.put("disabledClearAll", _disabledClearAll);
+		props.put("helpText", _helpText);
+		props.put("inputName", _inputName);
+		props.put("inputValue", _inputValue);
+		props.put("isValid", _valid);
+		props.put("label", _label);
+
+		if (_locator != null) {
+			props.put("locator", _locator);
+		}
+
+		props.put("selectedItems", _selectedItems);
+		props.put("sourceItems", _sourceItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("form-group");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		String id = getId();
+
+		if (_label != null) {
+			jspWriter.write("<label");
+
+			if (Validator.isNotNull(id)) {
+				jspWriter.write(" for=\"");
+				jspWriter.write(id);
+				jspWriter.write("\"");
+			}
+
+			jspWriter.write(">");
+
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<div class=\"input-group\"><div class=\"");
+		jspWriter.write("input-group-item\"><div class=\"form-control ");
+		jspWriter.write("form-control-tag-group input-group\"><div class=\"");
+		jspWriter.write("input-group-item\">");
+
+		List<Map<String, String>> selectedItems = getSelectedItems();
+
+		if (!ListUtil.isEmpty(selectedItems)) {
+			for (Map<String, String> selectedItem : selectedItems) {
+				String selectedItemLabel = selectedItem.get("label");
+
+				LabelTag labelTag = new LabelTag();
+
+				labelTag.setDismissible(true);
+				labelTag.setLabel(selectedItemLabel);
+
+				labelTag.doTag(pageContext);
+
+				jspWriter.write("<input type=\"hidden\" name=\"");
+				jspWriter.write(selectedItemLabel);
+				jspWriter.write("\" ");
+				jspWriter.write("value=\"");
+				jspWriter.write(selectedItem.get("value"));
+				jspWriter.write("\" />");
+			}
+		}
+
+		jspWriter.write("<input class=\"form-control-inset\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(id)) {
+			super.writeIdAttribute();
+		}
+
+		jspWriter.write(" type=\"text\" /></div>");
+
+		if (!_disabled && !_disabledClearAll && !selectedItems.isEmpty()) {
+			jspWriter.write("<div class=\"input-group-item ");
+			jspWriter.write("input-group-item-shrink\">");
+
+			ButtonTag buttonTag = new ButtonTag();
+
+			buttonTag.setCssClass("component-action");
+			buttonTag.setDisplayType("");
```

Comment:
This bugs me a bit. Clay's Button doesn't accept empty string as a display type value. I know it shouldn't be a problem here since this is just for SSR purposes, but it still feels weird.

- Can we use `unstyled`?
- If so, can we open an issue in Clay so we can replace [this](https://github.com/liferay/clay/blob/master/packages/clay-multi-select/src/index.tsx#L360) button with a `ClayButtonWithIcon` so we can have more consistency?

Reply:
This is how it looks like happens it we pass `"unstyled"` as the `displayType`

![](https://user-images.githubusercontent.com/5572/102100443-b79c7f00-3e29-11eb-8282-ecbf66c7298e.png)

And just so you know (and probably because you where going to ask too), this is what happens if we pass `"secondary"` as `displayType`

![](https://user-images.githubusercontent.com/5572/102100461-bc613300-3e29-11eb-9e4f-ed7392dc0000.png)

Probably because of this CSS rule

![](https://user-images.githubusercontent.com/5572/102101071-7a84bc80-3e2a-11eb-8c88-8f538fd7099d.png)

(Which I mentioned to @marcoscv-work)

I'm going to ask in clay, but I guess we could also render a button (i.e. not use `ButtonTag`)

Reply:
[Question asked](https://github.com/liferay/clay/issues/3850)

Reply:
I have tested this by making a change in Clay and linking the package locally, after replacing the `<button>` with `<ClayButtonWithIcon>` the Multi Select in DXP looks the same as it does on clayui.com